packages/allure-mocha/test/utils/index.ts

```diff
@@ -1,98 +1,27 @@
 import jetpack = require("fs-jetpack");
 import * as path from "path";
 import Mocha from "mocha";
-import propertiesReader from "properties-reader";
 import { MochaAllureReporter } from "../../src/MochaAllureReporter";
+import { InMemoryAllureWriter, TestResult } from "allure-js-commons";
 
 const testDir = "./dist/test/fixtures/specs";
-const resultsDir = path.join(__dirname, "../../../out/allure-results");
-const resultsPollingInterval: number = 10;
 
-let results: any[] = [];
-
-export function runTests(...specs: string[]) {
+export function runTests(...specs: string[]): Promise<InMemoryAllureWriter> {
+  const writer = new InMemoryAllureWriter();
   const mocha = new Mocha();
+  mocha.reporter(MochaAllureReporter, { writer });
```

Comment:
Here are the changes in a nutshell. Now it is possible to provide a custom writer implementation via options and redirect all output somewhere else.